### ISSUE TITLE
[Feature request] The ability to convert games, DLC, and Updates into a single file similar to how Cemu does with WUA on Wii U games.

### ISSUE THREAD

**Issue by TomTurbine** [RPCS3/rpcs3#15966]
### Quick summary

Basically games are stuck being divided up into disc based games which have to be decrypted in 1 folder while digital games have to be installed with keys in another along with updates.

My suggestion is basically just to be able to convert them into a single file type that the emulator can read while compressed similar to how Cemu can convert a game, its DLC, and its updates into a single WUA file. 

### Details

This would make it much easier to keep games and their files sorted and allow you to put them into a single folder while keeping them organized and easy to manage and easy for a front end to detect easily without jumping through a bunch of hoop, all while saving space.

**Comment by TomTurbine** [comment]
And to the people who dislike it, I have to ask why? Your games aren't kept as archive quality while playing them, you keep those copies stored elsewhere.

This would allow you to use your games you already extracted and decrypted and aren't archive quality anymore while keeping them more organized and saving on space all at the same time.

**Comment by elad335** [comment]
A single folder - one entry in file explorer view, able to be sorted by name
A single file - one entry in file explorer view, able to be sorted by name
How is a single file more organized?

**Comment by TomTurbine** [comment]
A single file that has the game, its DLC, and updates all included as that single file and takes up less space.
A single folder for the base game in the games folder plus the updates and dlc in another folder as loose files taking up more space.
How is that NOT more organized?

Having a single game in a single compressed file is far more organized than having the game and its other files stored in multiple other places.

This also makes it easier if you have your emulator on one drive while your games are on another.

**Comment by Megamouse** [comment]
Since installed game data is not read only, this makes no proper sense, and implementing this would create tons of side effects and complex issues that are just not worth the trouble.
Just thinking about how many problems this may add, makes me dizzy.

Also, the only application I can think of that would profit from this is file distribution.
And I'm pretty sure no-one in the RPCS3 team wants to play into the hands of piracy.

**Comment by TomTurbine** [comment]
I didn't think the emulator wrote files into the games DLC and Update folders past installing them.

And I wasn't thinking of Piracy, I was thinking more along the lines of organizing them, saving space, and its nice to have the applications running from a SSD for the added speed while the games can run off a slower conventional hard drive since they don't need the speed and the old spinner is much cheaper for space.

**Comment by Megamouse** [comment]
I don't think I have to explain how a single file is easier to distribute than having more files that need to be installed properly

**Comment by TomTurbine** [comment]
Your right, a single zip file is very easy to distribute regardless of how many files are in it.

And when it comes to Wii U piracy, do you see WUA files all over the place now? No, the places that serve that stuff try and serve the archive quality stuff.

If they created a .PS3 file type that worked the same way, you still wouldn't see them being served up like that, they would still stick to the archive quality stuff people can audit and feel secure it wasn't corrupted and is exactly what it is named.

**Comment by Megamouse** [comment]
you seem to know a lot about wii u piracy

**Comment by TomTurbine** [comment]
No, I know I setup my nephews Wii U last year for him on an external hard drive and researched stuff back then on how to mod his system and copies his games to it.

But nice attempt to "Gotcha" the post to avoid what I just pointed out. And besides, currently playing Baldur's Gate 3 which I purchased from GoG and Conquest of Azeroth which is free.

**Comment by marosis** [comment]
> No, I know I setup my nephews Wii U last year for him on an external hard drive and researched stuff back then on how to mod his system and copies his games to it.
> 
> But nice attempt to "Gotcha" the post to avoid what I just pointed out. And besides, currently playing Baldur's Gate 3 which I purchased from GoG and Conquest of Azeroth which is free.

This is not DIscord server

**Comment by Jack54guythecoder** [comment]
So will this be closed and not worked on any further?

**Comment by TomTurbine** [comment]
> This is not DIscord server
Understood, so he should stick to the merits of the request and issues around it and not make accusations like he just did, then I wouldn't have had to respond to it.

**Comment by GreemDev** [comment]
> Understood, so he should stick to the merits of the request and issues around it and not make accusations like he just did, then I wouldn't have had to respond to it.

He has and you just fail to understand why it won't be added.

**Comment by dcon03** [comment]
> So will this be closed and not worked on any further?

Most likely. The developers here have already made their point.

**Comment by Jack54guythecoder** [comment]
But could we possibly be able to read PKGs directly, without the need for them to be converted?
As in, being able to use PKG file directly.

**Comment by FlexBy420** [comment]
PKG files are essentialy archive file simillar to 7z, zip etc. they have to decrypted and extracted.

**Comment by TomTurbine** [comment]
>> He has and you just fail to understand why it won't be added.

No, I had just responded his response so he hadn't stopped at that point. And if it gets added or not, is fine by me. As far as WHY it isn't getting implemented, so be it. But if he claims "Piracy" he is giving a cop out and not paying attention to how that stuff actually works.

So, he can say, "We aren't planning on it right now", but a claim "Piracy" is a cop out because he isn't paying attention to how those places work.

**Comment by Jack54guythecoder** [comment]
> PKG files are essentialy archive file simillar to 7z, zip etc. they have to decrypted and extracted.

Oh, so its not something that can be just read like that.

**Comment by TomTurbine** [comment]
As for WHY I posted it here, you can partially blame Github for not allowing us to remove repositories years after we have used them and that idea crossing my mind and thinking of a way to improve the emulator.

**Comment by Megamouse** [comment]
I only added the piracy argument as an afterthought after pointing out the technical reasons, you were the one who focused on it so adamantly. I apologize if I offended you with the troll comment.

Anyway this is getting off-topic.

**Comment by TomTurbine** [comment]
Understood, I don't mean to take it too personal but have too many people who do that IRL when they try and avoid stuff or blame people for stuff.

As far as this, was just thinking of a nice feature to add for space saving and organization.

I'm out, later and best of luck.
